src/pages/Room/Room.js

```diff
@@ -25,16 +25,23 @@ class Room extends Component {
     socket.on('connect', () => {
       this.socketId = socket.id;
     });
-    socket.on('stateUpdate', (response) => {
+    socket.on('stateUpdate', (response, shouldUpdateMe) => {
       const me = response.team.find(client => client.id === this.socketId);
       const team = response.team.filter(client => client.id !== this.socketId);
       const show = response.show;
 
-      this.setState({
-        me,
+      this.setState(prevState => ({
+        /*
+        * For `Show` and `Clear` actions, we should also update `me` state
+        * to sync up with Redis data.
+        *
+        * For other actions, there is no need, since we don't want the scores
+        * will send to clients, so we reserve previous state for `me`.
+        */
+        me: shouldUpdateMe ? me : prevState.me,
```

Comment:
Good idea!

For the code you attached, I also set score to `me`, b/c we will use it for `Summary` component.

Please review the new commits I pushed.